wezterm-gui/src/termwindow/palette.rs

```diff
@@ -664,9 +664,14 @@ impl Modal for CommandPalette {
             .expect("to resolve char selection font");
         let metrics = RenderMetrics::with_font_metrics(&font.metrics());
 
-        let max_rows_on_screen = ((term_window.dimensions.pixel_height * 8 / 10)
+        let mut max_rows_on_screen = ((term_window.dimensions.pixel_height * 8 / 10)
             / metrics.cell_size.height as usize)
             - 2;
+        if 0 < term_window.config.command_palette_rows
+            && term_window.config.command_palette_rows < max_rows_on_screen
+        {
+            max_rows_on_screen = term_window.config.command_palette_rows;
+        }
```

Comment:
Can be simplified to:

```suggestion
        if let Some(size) = term_window.config.command_palette_rows {
            max_rows_on_screen = max_rows_on_screen.min(size);
        }
```